### DISCUSSION TITLE
shell integration with `fish` is not working

### DISCUSSION THREAD

**Discussion by ca057** [ghostty-org/ghostty#3439] [Q&A] [answered]
I’ve looked through the past issues regarding the shell integration with `fish`, but can’t figure out why the shell integration is not working in my case and how the previous solutions do apply in my case.

When I check the source code for the shell integration in fish, I see that I’m lacking the environment variable `GHOSTTY_SHELL_INTEGRATION_XDG_DIR` and the script returns early.

<img width="705" alt="Image" src="https://github.com/user-attachments/assets/a9f2afab-bb19-4d11-b1d0-a0571e8769b8" />

I’ve installed fish via mac ports and have the latest version. My fish setup should be relatively simple, I have some basic plugins installed via fisher.

Is there any more information I can provide to debug / understand this issue? Is there anything I’m missing after the installation to make it work?

**Comment by jparise** [comment]
Does setting `shell-integration = fish` help? If it does, then we'll know the problem is in the fish auto-detection path.

**Comment by gpanders** [comment, marked as answer]
The fact that you don't see `$GHOSTTY_SHELL_INTEGRATION_XDG_DIR` defined is an indication that shell integration _is_ working, since the shell integration script deletes that variable.

I also see from your screenshot that your cursor is a bar, which is another hint that shell integration is working.

What makes you think it isn't working?

**Reply by ca057** [reply]
Thanks for clarifying that it is working - this is an embarrassing oversight on my side after reading the feature list on the documentation 🙈 Apologies for opening up that issue way too quickly, I didn’t dive deep enough to understand the issue fully.

I mainly got confused by the `ghostty` cli command not being available and assumed it must be related to the shell integration being not properly configured. The `ghostty` command is not in my path, but I’ll try and understand first what is all happening to my path to prevent it from getting added.
